src/integrationtest/java/com/ericsson/ei/frontend/CommonSteps.java

```diff
@@ -139,6 +140,16 @@ public void request_sent() throws Throwable {
         response = httpRequest.performRequest();
     }
 
+    @When("^request is sent til body \'(.*)\' is not received$")
+    public void request_sent_body_not_received(String received) throws Throwable {
+        response = httpRequest.performRequest();
+        long stopTime = System.currentTimeMillis() + 30000;
+        while (response.getBody().equalsIgnoreCase(received) && stopTime > System.currentTimeMillis()) {
+            response = httpRequest.performRequest();
```

Comment:
This could be a do while loop so you dont have to add the performRequest line 2 times.